tba-unit-tests/TBAViewControllerTester.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import UIKit
+import XCTest
+
+class TBAViewControllerTester<T: UIViewController> {
+
+    private(set) var window: UIWindow!
+
+    init(withViewController rootViewController: T) {
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window.isHidden = false
+        window.rootViewController = rootViewController
+
+        if rootViewController is UINavigationController {
+            let navigationController = rootViewController as! UINavigationController
+            appearViewLifecyle(navigationController)
```

Comment:
Do we need to call this, or will the VC call this for us?

Reply:
It will not - although, I'm not sure how important driving that navigation controller lifecycle is

Reply:
Ah - it turns out our window will drive these for the root vc. We'll change the order of these so we're explicitly driving our lifecycle methods when testing